code/datums/components/food/edible.dm

```diff
@@ -455,18 +454,16 @@ Behavior that's still missing from this component that original food items had t
 	SIGNAL_HANDLER
 	SEND_SIGNAL(parent, COMSIG_FOOD_CROSSED, user, bitecount)
 
-///Add food types as integer containing food type flags.
-/datum/component/edible/proc/add_foodtypes(datum/source, type)
+///Response to customization, inherit tastes and foodtypes if possible
+/datum/component/edible/proc/on_customized(datum/source, obj/item/I)
 	SIGNAL_HANDLER
-	foodtypes |= type
 
-///Add new taste(s) as a string or list of strings.
-/datum/component/edible/proc/add_taste(datum/source, taste)
-	SIGNAL_HANDLER
-	if (islist(taste))
-		// union the list
-		for (var/s in taste)
-			if (!(s in tastes))
-				tastes.Add(s)
-	else if (!(taste in tastes))
-		tastes.Add(taste)
+	var/datum/component/edible/E = I.GetComponent(/datum/component/edible)
+	if (E)
+		/* tastes acting weird
+		if (LAZYLEN(E.tastes))
+			for (var/t in E.tastes)
+				tastes[t] += E.tastes[t]
+		*/
```

Comment:
Make sure to not leave commented out code in the final version.

Reply:
Working on a fix for the tastes